src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
```

Comment:
But from what I see here, it does require to await anyway, so the method can't be synchronous.
Also, using FromException is hit and miss with regards to preserving a stack trace, never had good experiences with that.
Having said all that, if there was a chance to make this method sync (which initially it doesn't look like), throwing ArgumentException is the only permitted exception to be thrown synchronously according to TAP design guidelines, as it is considered a [usage error](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap#exceptions) that shouldn't be handled, but the code should be fixed instead.